joblib/test/test_func_inspect.py

```diff
@@ -81,58 +62,61 @@ def f(self, x):
 ###############################################################################
 # Tests
 
-def test_filter_args():
-    yield assert_equal, filter_args(f, [], (1, )), {'x': 1, 'y': 0}
-    yield assert_equal, filter_args(f, ['x'], (1, )), {'y': 0}
-    yield assert_equal, filter_args(f, ['y'], (0, )), {'x': 0}
-    yield assert_equal, filter_args(f, ['y'], (0, ), dict(y=1)), {'x': 0}
-    yield assert_equal, filter_args(f, ['x', 'y'], (0, )), {}
-    yield assert_equal, filter_args(f, [], (0,), dict(y=1)), {'x': 0, 'y': 1}
-    yield assert_equal, filter_args(f, ['y'], (), dict(x=2, y=1)), {'x': 2}
 
-    yield assert_equal, filter_args(i, [], (2, )), {'x': 2}
-    yield assert_equal, filter_args(f2, [], (), dict(x=1)), {'x': 1}
+@parametrize(['funcname', 'args', 'filtered_args'],
+             [('f', [[], (1, )], {'x': 1, 'y': 0}),
```

Comment:
Why not use functions in parametrize?

```py
@parametrize(['func', 'args', 'filtered_args'],
    [(f, [[], (1, )], {'x': 1, 'y': 0}),
    ...
```

I guess the only good reason to use a fixture for funcs is actually the cached function `f2`, because it depends on tmpdir, right?